sw/device/lib/dif/dif_padctrl.h

```diff
@@ -0,0 +1,157 @@
+} dif_padctrl_init_result_t;
+
+/**
+ * Initialises an instance of Padctrl.
+ *
+ * Information that must be retained, and is required to program Padctrl, shall
+ * be stored in @p padctrl.
+ *
+ * @param base_addr Base address of an isntance of the Padctrl IP block
+ * @param padctrl Padctrl state data
+ * @return dif_padctrl_init_result_t.
+ */
+dif_padctrl_init_result_t dif_padctrl_init(mmio_region_t base_addr,
+                                           dif_padctrl_t *padctrl);
+
+/**
+ * Locks the padctrl registers.
+ *
+ * After the registers have been locked, they can only be unlocked by the
+ * system reset.
+ *
+ * @param padctrl Padctrl state data.
+ * @return dif_padctrl_result_t.
+ */
+dif_padctrl_result_t dif_padctrl_lock(dif_padctrl_t *padctrl);
+
+/**
+ * Padctrl attribute enable routines error codes.
+ */
+typedef enum dif_padctrl_enable_attr_result {
+  kDifPadctrlEnableOk = kDifPadctrlOk,
+  kDifPadctrlEnableError = kDifPadctrlError,
+  kDifPadctrlEnableBadArg = kDifPadctrlBadArg,
+  kDifPadctrlEnableAttrNotSupported, /**< Attribute is not supported. A write
+                                        has been performed to the attribute
+                                        registers but it had no effect as the
+                                        attribute isn't supported. Enable can
+                                        be retried with a different
+                                        attribute. */
+  kDifPadctrlEnableAttrConflict,     /**< Could not enable attribute because it
+                                        conflicts with another already enabled
+                                        attribute. No write to the attribute
+                                        registers has been attemped. Enable can
+                                        be retried with a different attribute. */
+} dif_padctrl_enable_attr_result_t;
+
+/**
+ * Enables or disables an attribute for a pad.
+ *
+ * Not all pads implement all attributes and some combinations of attributes
+ * cannot be enabled together. `dif_padctrl_enable_attr` will check for
+ * this and return an appropriate error. Note that for additional attributes as
+ * the meaning is implementation defined `dif_padctrl_enable_attr` cannot
+ * check for invalid combinations that involve them.
+ *
+ * @param padctrl Padctrl state data.
+ * @param pad_kind Which kind of pad to enable to an attribute for.
+ * @param pad_index Which pad to enable an attribute for.
+ * @param attr Attribute to enable.
+ * @param enable Enables if true, disabled otherwise.
+ * @return dif_padctrl_enable_attr_result_t
+ */
+dif_padctrl_enable_attr_result_t dif_padctrl_enable_attr(
+    const dif_padctrl_t *padctrl, dif_padctrl_pad_kind_t kind,
+    dif_padctrl_pad_index_t index, dif_padctrl_attr_t attr, bool enable);
+
+#endif  // OPENTITAN_SW_DEVICE_LIB_DIF_PADCTRL_H_
+
+#ifdef __cplusplus
+}  // extern "C"
+#endif  // OPENTITAN_SW_DEVICE_LIB_DIF_DIF_PADCTRL_H_
```

Comment:
I think you've got these mixed up? Should be the `extern "C"` bit first, followed by `#endif  // __cplusplus`.